envoy/stream_info/stream_info.h

```diff
@@ -626,6 +624,12 @@ class StreamInfo {
   virtual DownstreamTiming& downstreamTiming() PURE;
   virtual OptRef<const DownstreamTiming> downstreamTiming() const PURE;
 
+  /**
+   * @return the upstream timing information.
+   */
+  virtual UpstreamTiming& upstreamTiming() PURE;
```

Comment:
I am wondering why we are moving those two functions from `UpstreamInfo` to `StreamInfo` here? 

UpstreamInfo class is a good place for upstream related stuff like this. Your PR doesn't seem to require this move.

Reply:
Changed.